test/unit/custom_spec.js

```diff
@@ -107,3 +107,104 @@ describe("custom canvas rendering", function () {
       .catch(done.fail);
   });
 });
+
+describe("alternate document context", function () {
+  const FontFace = global.FontFace;
+
+  let CanvasFactory;
+
+  beforeAll(() => {
+    global.FontFace = function MockFontFace(name) {
+      this.family = name;
+    };
+    CanvasFactory = isNodeJS ? new NodeCanvasFactory() : new DOMCanvasFactory();
+  });
+
+  afterAll(() => {
+    global.FontFace = FontFace;
+    CanvasFactory = null;
+  });
+
+  it("should use given document for loading fonts", async function () {
+    const domDoc = { fonts: new Set() };
+    const getDocumentParams = buildGetDocumentParams(
+      "TrueType_without_cmap.pdf",
+      {
+        disableFontFace: false,
+        ownerDocument: domDoc,
+      }
+    );
+
+    const loadingTask = getDocument(getDocumentParams);
+    const doc = await loadingTask.promise;
+
+    const page = await doc.getPage(1);
+
+    const viewport = page.getViewport({ scale: 1 });
+    const canvasAndCtx = CanvasFactory.create(viewport.width, viewport.height);
+
+    await page.render({
+      canvasContext: canvasAndCtx.context,
+      viewport,
+    }).promise;
+
+    expect(domDoc.fonts.size).toBe(1);
+    const [font] = Array.from(domDoc.fonts);
+    console.dir(font);
+    expect(font.family).toMatch(/g_d[0-9a-f]+_f1/);
```

Comment:
This regular expression, and the one below, isn't entirely correct since the `docId` is always a number and nothing else. Hence you should drop the `a-f` part, and this can be simplified to `/g_d\d+_f1/` (similarily below).

Reply:
I didn't find the source of that string, so I went with a safe-ish pattern. I'm trying to tread lightly and respectfully with my changes. 😊 

I'll simplify the patterns.